pytensor/link/jax/dispatch/sparse.py

```diff
@@ -0,0 +1,38 @@
+import jax.experimental.sparse as jsp
+from scipy.sparse import spmatrix
+
+from pytensor.link.jax.dispatch import jax_funcify, jax_typify
+from pytensor.sparse.basic import Dot, SparseVariable, StructuredDot
+from pytensor.sparse.sharedvar import SparseTensorSharedVariable
+from pytensor.sparse.type import SparseTensorType
+
+
+@jax_typify.register(spmatrix)
+def jax_typify_spmatrix(matrix, dtype=None, **kwargs):
+    # Note: This changes the type of the constants from CSR/CSC to BCOO
+    # We could add BCOO as a PyTensor type but this would only be useful for JAX graphs
+    # and it would break the premise of one graph -> multiple backends.
```

Comment:
In order to not break the premise of one graph -> multiple backends, would there need to be extensions to the `SparseTensor` type, like `SparseCSRTensor`, `SparseCOOTensor` etc?

Reply:
We already have those, they wrap the Scipy equivalents, but JAX won't accept them as inputs to a jitted function.

I explored implementing a new SparseCOOTensor, but you still couldn't use the same object for JAX and C for instance, because JAX BCOO is a specific thing on its own, it's not a vanilla scipy COO_matrix.

So either we change everything else to accept JAX BCOO matrices, or users have to create graphs that use them in the first place, in which case they are creating a JAX-specific graph from the get go.

That's the crux of the issue. There must be a solution but it hasn't occurred to me yet. This is the same with RNG variables (jax uses PRNGs, and the rest of PyTensor uses numpy generators)...

Reply:
That's in a sense what we're doing here for constants.

For user inputs my previous prototype did something like that. I have to investigate more... it seemed too fragile/inneficient.